simulator/examples/clock.rs

```diff
@@ -0,0 +1,196 @@
+//!
+//! This example shows some more advanced usage of Embedded Graphics. It draws a round clock face
+//! with hour, minute and second hands. A digital clock is drawn in the middle of the clock. The
+//! whole thing is updated with your computer's local time every 50ms.
+
+use chrono::{Local, Timelike};
+use core::f32::consts::{FRAC_PI_2, PI};
+use embedded_graphics::egcircle;
+use embedded_graphics::fonts::Font12x16;
+use embedded_graphics::pixelcolor::BinaryColor;
+use embedded_graphics::prelude::*;
+use embedded_graphics::primitives::{Circle, Line, Rectangle};
+use embedded_graphics_simulator::DisplayBuilder;
+use std::thread;
+use std::time::Duration;
+
+const DISP_SIZE: i32 = 256;
+const CENTER: Point = Point::new(DISP_SIZE / 2, DISP_SIZE / 2);
+const SIZE: i32 = 120;
+
+/// Start at the top of the circle
+const START: f32 = -FRAC_PI_2;
+
+/// Draw a circle and 12 tics as a simple clock face
+fn draw_face() -> impl Iterator<Item = Pixel<BinaryColor>> {
+    let tic_len = 10.0;
+
+    // Use the circle macro to create the outer face
+    let face = egcircle!(
+        CENTER,
+        SIZE as u32,
+        stroke = Some(BinaryColor::On),
+        stroke_width = 2
+    );
+
+    // Create 12 `Line`s starting from the outer edge and drawing inwards by `tic_len` pixels
+    let tics = (0..12).into_iter().map(move |index| {
+        // Start angle around the circle, in radians
+        let angle = START + (PI * 2.0 / 12.0) * index as f32;
+
+        // Start point on circumference
+        let start = CENTER
+            + Point::new(
```

Comment:
Factoring out this code into a function like:
```rust
fn polar(angle: f32, radius: f32) -> Point {
    CENTER + Point::new(
        (angle.cos() * radius) as i32,
        (angle.sin() * radius) as i32,
    )
}
```
could remove some repeated code:
```rust
let start = polar(angle, SIZE as f32);
let end = polar(angle, SIZE as f32 - tic_len);
```